lib/pure/os.nim

```diff
@@ -2638,7 +2637,7 @@ proc parseCmdLine*(c: string): seq[string] {.
         while i < c.len and c[i] > ' ':
           add(a, c[i])
           inc(i)
-    add(result, a)
+    add(result, move(a))
```

Comment:
is that even correct to move something more than once (and should that even compile)?
```nim
proc main(): seq[string] =
  var a = "abc"
  for i in 0..<3: result.add move(a)
echo main()
```
this prints: `@["abc", "", ""]`

Reply:
It is expected behaviour I think.

Reply:
Original code
```nim
var a: string
while true:
  a.setLen(0)
  do_something_with(a)
  result.add a
```

After this PR
```nim
var a: string
while true:
  do_something_with(a)
  result.add move(a)
```

Reply:
the new code seems worse, at least from performance standpoint:
before PR, a single buffer is reused (`a`), and after each iteration `result.add a` just copies (hopefully eventually via memCopy under the hood) a.len bytes even if a has capacity > a.len.

after PR, a fresh buffer is created at each new iteration, and each iteration can cause a buffer resizes with capacity exceeding the length.